source/game/kart/KartMove.cc

```diff
@@ -1593,6 +1624,146 @@ void KartMove::landTrick() {
     activateBoost(KartBoost::Type::TrickAndZipper, duration);
 }
 
+/// @addr{0x8058498C}
+void KartMove::enterCannon() {
+    init(true, true);
+    physics()->clearDecayingRot();
+    m_boost.resetActive();
+    state()->setBoost(false);
+
+    cancelJumpPad();
+    m_rampBoost = 0;
+    state()->setRampBoost(false);
+    m_ssmtCharge = 0;
+    m_bPadBoost = false;
+    m_bRampBoost = false;
+    m_bPadJump = false;
+    m_bSsmtCharged = false;
+    m_bSsmtLeeway = false;
+    m_bTrickableSurface = false;
+    m_bWallBounce = false;
+    m_offroadInvincibility = 0;
+    state()->setBoostOffroadInvincibility(false);
+
+    dynamics()->reset();
+
+    state()->setHop(false);
```

Comment:
```suggestion
    clearDrift();
    state()->setHop(false);
```

Reply:
mmkay clearly there's some unlabeled stuff in my ghidra: 
```cpp
  pKVar7->bitfield1 = pKVar7->bitfield1 | IN_CANNON;
  pKVar7 = ((this->inherit).mpPointers)->state;
  pKVar7->bitfield2 = pKVar7->bitfield2 | SKIP_WHEEL_CALC;
  pKVar7 = ((this->inherit).mpPointers)->state;
  pKVar7->bitfield1 = pKVar7->bitfield1 & ~CANNON_START;
  this->field110_0x1e0 = 0.0;
  KartObjectProxy::getCannonPosAndDir(&this->inherit,&local_1c,&vStack_28);
  this->field111_0x1e4 = 0.0;
```
mind clarifying what `clearDrift()` is? do you mean `0x8057EA50 void KartMove::resetDriftManual()`?

Reply:
See the `bl` at 80584b58.